platform/security/sts/security-sts-server/src/main/java/ddf/security/sts/MemoryTokenStore.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.security.sts;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import java.io.Closeable;
+import java.util.Collection;
+import java.util.concurrent.TimeUnit;
+import org.apache.cxf.buslifecycle.BusLifeCycleListener;
+import org.apache.cxf.ws.security.tokenstore.SecurityToken;
+import org.apache.cxf.ws.security.tokenstore.TokenStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MemoryTokenStore implements TokenStore, Closeable, BusLifeCycleListener {
+
+  private static final Cache<String, SecurityToken> CACHE =
+      CacheBuilder.newBuilder().expireAfterWrite(15, TimeUnit.MINUTES).maximumSize(1000).build();
```

Comment:
How do we determine the expire time?

Reply:
15 minutes was chosen as it's just a cache to keep from hitting LDAP too often in those user/password login cases. I also arbitrarily chose to limit the cache size to 1000 to keep from consuming too much memory. The previous cache was not configured via metatype, so I didn't add configuration for this.

Reply:
Should be fine. This cache is really just here to slow down requests to an attribute store. We could even lower it.

Reply:
I'll probably leave the 15 how it is since the Unit goes with it, and this is the only place the constant is used.

Reply:
I think we still want to have this data expire after a set amount of time (vs staying in cache perpetually). @stustison is this correct, or should I change to use `expireAfterAccess`?